python/lsst/obs/base/formatters/fitsExposure.py

```diff
@@ -418,21 +431,98 @@ def checkUnrecognized(entry, allowed, description):
         return validated
 
 
-class FitsImageFormatter(FitsExposureFormatter):
+class FitsExposureFormatter(FitsImageFormatterBase):
+    """Specialization for `~lsst.afw.image.Exposure` reading.
+    """
+
+    _readerClass = ExposureFitsReader
+
+    def _compareFilters(self, filter_label):
+        """Compare the filter label read from the file with the one in the
+        data ID.
+
+        Parameters
+        ----------
+        filter_label : `lsst.afw.image.FilterLabel`, optional
+            Filter label read from the file, if there was one.
+
+        Returns
+        -------
+        filter_label : `lsst.afw.image.FilterLabel` or `None`
+            The preferred filter label; may be the given one or one built from
+            the data ID.
+        """
+        missing = []
+        band = None
+        physical_filter = None
+        if "band" in self.dataId.graph.dimensions.names:
+            band = self.dataId.get("band")
+            if band is None and not self.dataId.hasFull() and "band" in self.dataId.graph.implied.names:
+                missing.append("band")
+        if "physical_filter" in self.dataId.graph.dimensions.names:
+            physical_filter = self.dataId.get("physical_filter")
+            if (physical_filter is None and not self.dataId.hasFull()
+                    and "physical_filter" in self.dataId.graph.implied.names):
+                missing.append("physical_filter")
+        version = self._reader.readSerializationVersion()
+        if version < 2:
```

Comment:
I don't understand why you have two redundant branches that depend on an implementation detail of `afw`. Either the `FilterLabel` in the `Exposure` is consistent with the data ID, or it isn't. Why do you apply completely different tests on completely different variables?

Reply:
I did not consider the serialization version an implementation detail; I considered it the best way to find out whether the filter value(s) in the file are likely to be wrong or incomplete.  I'm happy to switch to a different approach if you can identify one.

And essentially "are the filter values in the file trustworthy" is what the branching here is about.  If the file values are likely to be wrong, we don't want to warn when they are, but we can fix them, but we do want to warn if we can't fix them.  If the file values are likely to be right, we warn if the data ID says they're wrong, and don't warn if we can't check them.

Reply:
FWIW, with the new logic (following the discussion), there is still a notion of trustworthiness derived from the serialization version here, but a much weaker one; it's now only used to eliminate warnings for pre-`FilterLabel` files.

Reply:
Can you explain the reasoning behind that behavior? At the very least, users should probably be warned that they're using old files -- especially since the quality of the translation will become _much_ worse once we remove `Filter`.

Reply:
As far as I can tell, all of DP0.1 will be using old files, and iwe may have HSC files that old in use for a while yet, too.  Anyone going through this formatter who does not get a warning will not care whether they get old files; the quality of the filter translation will be irrelevant because it'll be overwritten by what's in the data ID.  (Unless I misunderstood what you meant by "translation".)